[jinhanjinhan] iP

jinhanjinhan opened the pull request:
# Juno 
> “Your mind is for having ideas, not holding them.” – David Allen

## Overview

Juno is a chatbot that allows you to keep track of the current tasks you have.

All you need to do is,

1. **download** it from [here](https://github.com/jinhanjinhan/ip/releases/download/v0.1/juno.jar).
2. double-click it.
3. add your tasks.
5. let it manage your tasks for you 😉

Worry that your tasks might not be saved? Fret not!
The tasks added will be automatically saved each time you enter or delete a task, so even if you exit the chat, it's fine!

And most importantly, it is **FREE**!

---

Key Strengths:

- Easy to learn
- User friendly text based interface
- FAST!


Features:

- [x] Managing normal tasks
- [x] Managing deadlines
- [x] Managing special events
- [ ] Normal daily chatting (coming soon!)

-------------

> [!Note]
> New features are coming soon, Stay tuned!




---

If you are a **Java** programmer, you can use it to practice **Java** too. Here's an example of the `detectUserInput()` method:

```java
    /**
     * Detects user input through the command line.
     * User input is parsed into commands, which are then executed. The loop continues until
     * the user inputs a command "bye" to terminate the chat bot.
     */
    public void detectUserInput() {
        Scanner scanner = new Scanner(System.in);
        boolean isInWhileLoop = true;

        while (isInWhileLoop) {
            String userInput = scanner.nextLine().trim();
            try {
                Command command = this.commandParser.parse(userInput, this.junoUi, this.fileManager, this.taskManager);
                command.runCommand();
                isInWhileLoop = command.isInWhileLoop();
            } catch (TaskManagerException e) {
                System.out.println(e.getMessage());
            }
        }
        scanner.close();
    }
```
The `detectUserInput()` method, as the name suggests, will detect the input that the user provided and execute the relevant commands accordingly.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jinhanjinhan committed:
Renamed chatbot to Juno and added greet and exit functionality

jinhanjinhan committed:
Modified chatbot class, and added chatbot echo functionality

jinhanjinhan committed:
Added functionality for chatbot to store text and display when user requests

jinhanjinhan committed:
Added feature to mark tasks as done

jinhanjinhan committed:
Added TaskManager for all task-related operations and added tracking for 3 types of tasks

jinhanjinhan committed:
Added testing using the I/O redirection technique

jinhanjinhan committed:
Added TaskManagerException class to handle exceptions

jinhanjinhan committed:
Added delete feature

jinhanjinhan committed:
Task related classes: add taskType and getter

jinhanjinhan committed:
Git ignore: ignore *.jar files

jinhanjinhan committed:
Juno and TaskManager class: modify to incorporate new fileManager

jinhanjinhan committed:
TaskAdapter class: Customise read and write methods

jinhanjinhan committed:
FileManager class: add to control all file-related operations

jinhanjinhan committed:
Merge branch 'branch-Level-7'

jinhanjinhan committed:
Date feature: use DateTime class to store time instead of string

jinhanjinhan committed:
TextUiTexting changes: Modify input command and expected output

jinhanjinhan committed:
Merge branch 'branch-Level-8'

jinhanjinhan committed:
TaskManagerException changes: add new Exception types

jinhanjinhan committed:
TaskManager class: remove command specific methods

jinhanjinhan committed:
Juno* classes: add JunoUi for user interface related methods

jinhanjinhan committed:
Event class: modify Exception message for inputting invalid date

jinhanjinhan committed:
CommandParser class: add to interpret command input by users

jinhanjinhan committed:
Command-related classes: add to handle each type of command

jinhanjinhan committed:
Package: Organise classes into different packages

jinhanjinhan committed:
Merge branch 'add-gradle-support'

jinhanjinhan committed:
Gradle file: add in gson dependency

jinhanjinhan committed:
Exception related change: Add new TaskManagerException type to enum

jinhanjinhan committed:
Build: Add mockito dependency to build.gradle to mock dependencies

jinhanjinhan committed:
Test files: Add JUnit test for non-trivial methods

jinhanjinhan committed:
Build: Add properties to shadowJar

jinhanjinhan committed:
Javadoc comments: add Javadoc comments detailing usage of each class

jinhanjinhan committed:
Javadoc comments: modify to comply with Javadoc coding standards

jinhanjinhan committed:
Juno class: Add to the juno package

jinhanjinhan committed:
Coding standard fix: Fix to comply with Java Coding Standard

jinhanjinhan committed:
Javadoc: Add Javadoc comments to Juno and JunoUi class

jinhanjinhan committed:
Merge branch 'branch-A-Javadoc'

jinhanjinhan committed:
Coding standard fix: Fix to comply with Java coding standard

jinhanjinhan committed:
Find feature: Add find command which allows users to search from list

jinhanjinhan committed:
Find feature: Add find command which allows users to search from list

jinhanjinhan committed:
Checkstyle plugin: add checkstyle plugin

jinhanjinhan committed:
Checkstyle: modify classes according to style guide

zeonchew0324 commented on src/test/java/juno/command/AddDeadlineCommandTest.java:
```diff
@@ -0,0 +1,50 @@
+import juno.manager.exception.TaskManagerException;
+import juno.task.Deadline;
+import juno.task.Task;
+
```
Good job! You are consistent with the ordering of your import packages.

zeonchew0324 commented on src/main/java/juno/command/Command.java:
```diff
@@ -0,0 +1,29 @@
+ * Abstract class for all possible commands.
+ * Defines how commands can be executed.
+ * Superclass for all other *Command classes in juno.command package
+ */
```
Would not it be better to add enums at this command file?

zeonchew0324 commented on src/main/java/juno/Juno.java:
```diff
@@ -0,0 +1,90 @@
+import juno.parser.CommandParser;
+import juno.task.Task;
+import juno.ui.JunoUi;
+
```
Very clear javadoc explaination, well done!

zeonchew0324 commented on src/main/java/juno/manager/exception/TaskManagerException.java:
```diff
@@ -0,0 +1,45 @@
+        INVALID_DELETE_TASK_NUMBER,
+        NO_TASK_FOUND,
+        INVALID_FIND_TASK
+    }
```
Good that your names follow the coding naming standard!

zeonchew0324 reviewed (commented):
Looks good to me! Not much problem to merge.

jinhanjinhan committed:
GUI addition: Modify current CLI to GUI using JavaFX

jinhanjinhan committed:
Merge branch 'branch-Level-10'

ngjianyi commented on src/main/java/juno/command/MarkCommand.java:
```diff
@@ -0,0 +1,92 @@
+    private FileManager fileManager;
+    private String userInput;
+    private ArrayList<Task> tasks;
+    private boolean markAsDone;
```
Perhaps the name used can be made to sound more like a boolean? e.g. isMarkedAsDone

ngjianyi commented on src/main/java/juno/manager/TaskAdapter.java:
```diff
@@ -0,0 +1,116 @@
+        reader.beginObject();
+        String taskType = "";
+        String description = "";
+        boolean isDone = false;
```
I like the clear and concise names!

jinhanjinhan committed:
ListCommandTest class: Modify to fit the GUI implementation

The runCommand_nonEmptyList_success() test case in ListCommandTest class
fails.

The failure is caused by its implementation, which involves reading
output to the terminal. But since the current implementation is GUI,
there are no more outputs to the terminal.

Let's modify the method to read the returning string from the
runCommand() method for testing.

Doing this is preferrable as the string returned from runCommand() is
being passed and subsequently displayed on the GUI, so it is the same
one as the one seen from the GUI, ensuring that it is a valid test case.

jinhanjinhan committed:
Gradle file update: Add JavaFX dependencies for build

jinhanjinhan committed:
Java assert addition: Add assert statements to ensure correctness

The current implementation does not have assert statements.

This results in potential crashes in the code if certain assumptions are
not met.

Let's add some assert statements into the codebase to document
important assumptions that should hold at various points in the code.

This will ensure correctness of the code.

jinhanjinhan committed:
Java assert addition: Add assert statements to ensure correctness

The current implementation does not have assert statements.

This results in potential crashes in the code if certain assumptions are
not met.

Let's add some assert statements into the codebase to document
important assumptions that should hold at various points in the code.

This will ensure correctness of the code.

jinhanjinhan committed:
Revert "Java assert addition: Add assert statements to ensure correctness"

This reverts commit b625bbc0c86cf8ae0c9b7c9dee20a95bab705df3.

jinhanjinhan committed:
Merge pull request #1 from jinhanjinhan/branch-A-Assertions

Merge branch-A-Assertions into master

jinhanjinhan committed:
Merge branch 'master' into branch-A-Streams

jinhanjinhan committed:
Code quality check: Improve code quality from current files

jinhanjinhan committed:
Merge master into branch-A-CodeQuality

jinhanjinhan committed:
Merge pull request #3 from jinhanjinhan/branch-A-CodeQuality

Code quality check: Improve code quality from current files

jinhanjinhan committed:
Merge branch 'master' into branch-A-Streams

jinhanjinhan committed:
Streams addition: Add Java streams to applicable methods

jinhanjinhan committed:
Merge pull request #2 from jinhanjinhan/branch-A-Streams

Branch a streams

jinhanjinhan committed:
CI addition: Add CI workflow for Juno

jinhanjinhan committed:
CI changes: Modify gradle.yml to pass build

jinhanjinhan committed:
CI changes: Modify gradle.yml to pass build

jinhanjinhan committed:
Merge branch 'master' of https://github.com/jinhanjinhan/ip

jinhanjinhan committed:
Javadoc addition: Add Javadoc comments to FindCommand class

jinhanjinhan committed:
Task-related addition: Add getter methods to retrieve end time

jinhanjinhan committed:
Add* commands changes: Change TASK_TYPE to static field

jinhanjinhan committed:
Remind feature: Add a way for users to get reminders on scheduled tasks

jinhanjinhan committed:
Merge branch 'branch-BCD-extension'

jinhanjinhan committed:
GUI improvements: Change visual format for user's input

jinhanjinhan committed:
GUI title: Add title to Juno GUI

jinhanjinhan committed:
Docs changes: Modify README.md and add images for project

jinhanjinhan committed:
Git LFS: Add images using Git LFS

jinhanjinhan committed:
Revert "Git LFS: Add images using Git LFS"

This reverts commit 70dcf3e5ec145f04bac1ae4ac04282252b61a67e.

jinhanjinhan committed:
Docs folder changes: remove images in docs folder

jinhanjinhan committed:
Build file changes: Modify build.gradle to pass CI

jinhanjinhan committed:
MarkCommandTest class: Add a test method and remove dead code

jinhanjinhan committed:
Method length issues: Shorten methods to suit Java coding standard

jinhanjinhan committed:
Java Coding Standard check: Modify to adhere to coding standards

jinhanjinhan committed:
Docs modification: Modify size of images and add unmark documentation

jinhanjinhan committed:
Docs modification: Modify size of main GUI image

jinhanjinhan committed:
Docs modification: Modify notes section of feature

jinhanjinhan committed:
CommandParser class: Shorten method length for parse() method

jinhanjinhan committed:
MainWindow class: Add Javadoc comments to a method

jinhanjinhan committed:
Readme docs edit: Correct spelling and grammar errors

jinhanjinhan committed:
Exception handling: Edit error messages for clarity

jinhanjinhan committed:
Test classes: Update test classes

jinhanjinhan committed:
Docs edit: Correct grammar errors

jinhanjinhan committed:
Date formatter fix: Fix locale issues with DateTimeFormatter

jinhanjinhan committed:
Task classes: Add trim() to remove trailing whitespaces

jinhanjinhan committed:
Test methods fix: Correct expected outputs

jinhanjinhan committed:
Add*Command classes: Add trim() to remove whitespaces
